pkg/registration/spoke/spokeagent.go

```diff
@@ -244,12 +243,6 @@ func (o *SpokeAgentConfig) RunSpokeAgentWithSpokeInformers(ctx context.Context,
 	// in scenario #2 and #3, which results in an error message in log: 'Observed a panic: timeout waiting for
 	// informer cache'
 	if !ok {
-		// delete the hub kubeconfig secret if exits
```

Comment:
Should we add a note in line 290 that if the secret is not valid, the xxx controller will update it to a valid one?

Reply:
And maybe more note that waiting forever in line 290 is because the cluster should be approved on the hub side?

Reply:
Comments added as suggested.